### ISSUE TITLE
rpcdaemon crashes on certain eth_getLogs on devel

### ISSUE THREAD

**Issue by banteg** [ledgerwatch/erigon#9893]
#### System information

Erigon version: `erigon version 2.60.0-dev-70406e68`

OS & Version: Linux

Commit hash: 70406e6887e1a489daccc4964bbacb560e3598ee

Erigon Command (with flags/config):
```
rpcdaemon \
    --datadir /home/banteg/.local/share/erigon/mainnet \
    --http.api eth,net,web3,engine,debug,trace,erigon,parity,ots \
    --http.compression false \
    --metrics \
    --rpc.batch.concurrency 8 \
    --rpc.batch.limit 10000 \
    --rpc.gascap 0 \
    --rpc.returndata.limit 100000000
```

Consensus Layer: lighthouse

Consensus Layer Command (with flags/config):

Chain/Network: mainnet

#### Expected behaviour
an empty list is returned

#### Actual behaviour
rpcdaemon crashes

#### Steps to reproduce the behaviour
```
cast rpc eth_getLogs '{"address":"0x6B175474E89094C44Da98b954EedeAC495271d0F","fromBlock":10253600,"toBlock":10253605,"topics":["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef","0x0000000000000000000000000000000000000000000000000000000000000000"]}'

Error:
error sending request for url (http://127.0.0.1:8545/): connection closed before message completed

Context:
- connection closed before message completed
```

#### Backtrace
https://gist.github.com/banteg/14cc9193c7c78331e97b28b71dc695c3

**Comment by yperbasis** [comment]
@somnathb1 is this fixed by PR #10078?

**Comment by somnathb1** [comment]
> @somnathb1 is this fixed by PR #10078?

Haven't tested it, but that PR wasn't directed to fix this, but rather just an improvement

**Comment by banteg** [comment]
can't test this anymore, since i resynced the node. let's assume this is fixed since @somnathb1 has worked off my crash logs.
